### DISCUSSION TITLE
Switching from System.Drawing to SkiaSharp, need help with the classes mapping

### DISCUSSION THREAD

**Discussion by mizzPpY** [mono/SkiaSharp#2628] [Q&A]
Hi all,
I am new to SkiaSharp and currently working on a .NET 6 project and want to move from System.Drawing to SkiaSharp. May I know what is the SKiaSharp equivalent of _System.Drawing. Graphics_, and _System.Drawing.Imaging.ImageAttributes_? 

Many thanks!

**Comment by ojb500** [comment]
`System.Drawing.Graphics` is most similar to `SKSurface` in that it represents a surface that can be drawn onto by accessing `SKSurface.Canvas`

I believe there is no direct equivalent for `System.Drawing.Imaging.ImageAttributes`, this depends on what you are trying to achieve. Various adjustments can be done by passing an `SKPaint` to `canvas.DrawImage` and if you want to tile an image this can be done by making an `SKShader` and setting the `SKShaderTileMode`
